### PR TITLE
modules: liblc3: Rename the module for consistency

### PR DESCRIPTION
**Pull request by carlescufi**
The old project, liblc3codec, is not used anymore. Rename the Zephyr
module name, folders, and Kconfig options based on the new name, liblc3.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

### CODE REVIEW

**Review by tejlmand — DISMISSED** [review]

File: west.yml [hunk 1]
```diff
@@ -150,8 +150,8 @@ manifest:
       groups:
         - hal
     - name: liblc3
-      revision: 54c047249bc6b8089e85c1433b5aa6001bbc1ba9
-      path: modules/lib/liblc3codec
+      revision: pull/1/head
```
Comment (Thalley) [inline west.yml:153]: Is this intentional? Wouldn't that cause us to use the newest always, instead of a specific revision?

Comment (stephanosio) [reply west.yml:153]: This is still DNM

Comment (carlescufi) [reply west.yml:153]: No, this is the standard way to submit module updates:
https://docs.zephyrproject.org/latest/develop/modules.html#submitting-new-modules

But I now realized that the module.yml is actually upstream, so submitted:
https://github.com/google/liblc3/pull/8

**Review by Thalley — DISMISSED** [review]

**Comment by carlescufi** [comment]
@Thalley @stephanosio @Casper-Bonde-Bose @MariuszSkamra this is now ready, since the upstream PR is merged

**Review by stephanosio — APPROVED** [review]

**Review by Thalley — APPROVED** [review]

**Pull request merged**